src/configureWorkspace/configureDotNetCore.ts

```diff
@@ -249,14 +262,85 @@ function genDockerFile(serviceNameAndRelativePath: string, platform: Platform, o
         .replace(/\$assembly_name\$/g, assemblyNameNoExtension)
         .replace(/\$copy_project_commands\$/g, copyProjectCommands);
 
+    validateForUnresolvedToken(contents);
+
+    return contents;
+}
+
+function validateForUnresolvedToken(contents: string): void {
     let unreplacedToken = extractRegExGroups(contents, /(\$[a-z_]+\$)/, ['']);
     if (unreplacedToken[0]) {
         assert.fail(`Unreplaced template token "${unreplacedToken}"`);
     }
+}
 
-    return contents;
+function generateComposeFiles(dockerfileName: string, platform: Platform, os: PlatformOS | undefined, ports: number[], artifactName: string): ScaffoldFile[] {
+    const serviceName = path.basename(artifactName, path.extname(artifactName)).toLowerCase();
+    const imageName = serviceName;
+    let jsonPorts: string = '';
+    if (ports && ports.length > 0) {
+        const portsArray = ports.map(p => `      - ${p}`).join("\r\n");
+        jsonPorts = `
+    ports:
+${portsArray}`;
+    }
+
+    let environmentVariables: string = '';
+    if (platform === 'ASP.NET Core') {
+        environmentVariables = `    environment:
+      - ASPNETCORE_ENVIRONMENT=Development`;
+        // For now assume the first port is http and the second is https. (default scaffolding behavior)
+        // TODO: This is not the perfect logic, this should be improved later.
+        if (ports && ports.length > 0) {
+            // eslint-disable-next-line @typescript-eslint/tslint/config
+            let aspNetCoreUrl: string = `      - ASPNETCORE_URLS=http://+:${ports[0]}`;
+            if (ports.length >= 2) {
+                aspNetCoreUrl += `;https://+:${ports[1]}`;
+            }
+            environmentVariables += `\r\n${aspNetCoreUrl}`
+        }
+    }
+
+    let volumesList = '      - ~\\.vsdbg:/remote_debugger';
```

Comment:
Does this get added to both Compose files? We should make sure the debugger mapping only is added to the `docker-compose.debug.yml` file.

Reply:
Yes it is added only to debug.yml